packages/scaffold-config/src/frameworks.ts

```diff
@@ -0,0 +1,579 @@
+import {
+  CODE_GEN_FRAMEWORKS,
+  FRONTEND_FRAMEWORK_CATEGORIES,
+  CYPRESS_REACT_LATEST,
+  CYPRESS_VUE_2,
+  CYPRESS_VUE_3,
+  HTML_WEBPACK_PLUGIN_4,
+  STORYBOOK_REACT,
+  STORYBOOK_VUE,
+  WEBPACK_DEV_SERVER_4,
+  BUNDLER_WEBPACK_4,
+  CYPRESS_WEBPACK,
+} from './constants'
+import { BUNDLER_VITE, BUNDLER_WEBPACK_5, HTML_WEBPACK_PLUGIN_5 } from '.'
+
+const isWebpack = (bundler: Bundler) => ['webpack4', 'webpack5'].includes(bundler)
+
+export const FRONTEND_FRAMEWORKS = [
+  {
+    type: 'crav4',
+    family: 'template',
+    name: 'Create React App (v4)',
+    supportedBundlers: [BUNDLER_WEBPACK_4],
+    packages: [
+      CYPRESS_REACT_LATEST,
+      BUNDLER_WEBPACK_4,
+      CYPRESS_WEBPACK,
+      WEBPACK_DEV_SERVER_4,
+      HTML_WEBPACK_PLUGIN_4,
+    ],
+    defaultPackagePath: '@cypress/react/plugins/react-scripts',
+    glob: '*.{js,jsx,tsx}',
+    category: FRONTEND_FRAMEWORK_CATEGORIES[0],
+    codeGenFramework: CODE_GEN_FRAMEWORKS[0],
+    storybookDep: STORYBOOK_REACT,
+    detectors: [
+      {
+        dependency: 'react-scripts',
+        version: '^4.0.0',
+      },
+    ],
+    config: {
+      js: () => {
+        return dedent`
+      const { devServer } = require('@cypress/react/plugins/react-scripts')
+
+      module.exports = {
+        component: {
+          devServer,
+        }
+      }`
+      },
+      ts: () => {
+        return dedent`
+      import { defineConfig } from 'cypress'
+      import { devServer } from '@cypress/react/plugins/react-scripts'
+
+      export default defineConfig({
+        component: {
+          devServer,
+        }
+      })`
+      },
+    },
+  },
+  {
+    type: 'crav5',
+    family: 'template',
+    name: 'Create React App (v5)',
+    supportedBundlers: [BUNDLER_WEBPACK_5],
+    packages: [
+      CYPRESS_REACT_LATEST,
+      BUNDLER_WEBPACK_5,
+      CYPRESS_WEBPACK,
+      WEBPACK_DEV_SERVER_4,
+      HTML_WEBPACK_PLUGIN_5,
+    ],
+    defaultPackagePath: '@cypress/react/plugins/react-scripts',
+    glob: '*.{js,jsx,tsx}',
+    category: FRONTEND_FRAMEWORK_CATEGORIES[0],
+    codeGenFramework: CODE_GEN_FRAMEWORKS[0],
+    storybookDep: STORYBOOK_REACT,
+    detectors: [
+      {
+        dependency: 'react-scripts',
+        version: '^5.0.0',
+      },
+    ],
+    config: {
+      js: () => {
+        return dedent`
+      const { devServer } = require('@cypress/react/plugins/react-scripts')
+
+      module.exports = {
+        component: {
+          devServer,
+        }
+      }`
+      },
+      ts: () => {
+        return dedent`
+      import { defineConfig } from 'cypress'
+      import { devServer } from '@cypress/react/plugins/react-scripts'
+
+      export default defineConfig({
+        component: {
+          devServer,
+        }
+      })`
+      },
+    },
+  },
+  {
+    type: 'vueclivue2',
+    name: 'Vue CLI (Vue 2)',
+    family: 'template',
+    supportedBundlers: [BUNDLER_WEBPACK_4],
+    packages: [CYPRESS_VUE_2, CYPRESS_WEBPACK],
```

Comment:
Should we do the same thing here that we do with `cra` i.e. list all webpack deps?

Reply:
Done for `template` 107cd32209e574d683e53697ffab95035add9a35